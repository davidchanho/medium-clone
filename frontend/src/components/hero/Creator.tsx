import React from "react";
import { Button, ListGroupItem, ListGroupItemProps } from "react-bootstrap";
import { clampText } from "../../helpers/clampText";
import { IUser } from "../../types";

function Creator({
  photo,
  name,
  about,
  className,
  ...props
}: IUser & ListGroupItemProps) {
  return (
    <ListGroupItem
      className={`w-100 border-0 d-flex align-items-center justify-content-between ${className}`}
      {...props}
    >
      <img src={photo} alt="creator avatar" className="rounded-circle" width='60' height='60' />

      <div className="w-50 mx-2">
        <p className="font-weight-bold">{name}</p>
        <p>{about && clampText(about, 35)}</p>
      </div>

      <Button variant="success" className="h-50 my-auto">
        Follow
      </Button>
    </ListGroupItem>
  );
}

export default Creator;
