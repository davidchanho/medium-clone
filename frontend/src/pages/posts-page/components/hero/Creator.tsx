import React, { HTMLAttributes } from "react";
import { clampText } from "../../../../helpers/clampText";
import { IUser } from "../../../../types";

function Creator({
  photo,
  name,
  about,
  className,
  ...props
}: IUser & HTMLAttributes<HTMLLIElement>) {
  return (
    <li className={className} {...props}>
      <img src={photo} alt="creator avatar" width="60" height="60" />

      <div>
        <p>{name}</p>
        <p>{about && clampText(about, 35)}</p>
      </div>

      <button>Follow</button>
    </li>
  );
}

export default Creator;
