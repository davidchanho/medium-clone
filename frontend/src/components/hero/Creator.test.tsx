import React from "react";
import renderer from "react-test-renderer";
import { testUser } from "../../tests";
import Creator from "./Creator";

describe("Creator", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Creator user={testUser} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
