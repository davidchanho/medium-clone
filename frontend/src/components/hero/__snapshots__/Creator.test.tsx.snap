// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creator renders correctly 1`] = `
<div
  className="w-100 d-flex justify-content-between list-group-item"
  disabled={false}
  onClick={[Function]}
>
  <img
    className="rounded-circle"
    src="test"
  />
  <div
    className="w-50 mx-2"
  >
    <p
      className="font-weight-bold"
    >
      test
    </p>
    <p>
      test...
    </p>
  </div>
  <button
    className="h-50 my-auto btn btn-success"
    disabled={false}
    type="button"
  >
    Follow
  </button>
</div>
`;
